scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/ReplaceAfterTerminateScheduler.java

```diff
@@ -15,18 +15,35 @@
 package com.amazonaws.blox.scheduling.scheduler.engine.daemon;
 
 import com.amazonaws.blox.scheduling.scheduler.engine.SchedulingAction;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class ReplaceAfterTerminateScheduler extends DaemonScheduler {
   public static final String ID = "ReplaceAfterTerminate";
 
   public List<SchedulingAction> schedule(DaemonEnvironment env, ClusterSummary summary) {
-    return summary
-        .getInstances()
-        .stream()
-        .filter(i -> env.hasMatchingTask(summary.tasksForInstance(i)))
-        .map(i -> env.startTaskFor(i))
-        .collect(Collectors.toList());
+    List<ClusterSnapshot.Task> tasksToStop =
+        summary
+            .getInstances()
+            .stream()
+            .filter(i -> env.needsToStop(summary.tasksForInstance(i)))
```

Comment:
Not sure the filter on instance for `needsToStop` has value here. if the `env.taskToStop(t)` can find out all the task to stop, if there is no task in the instance, a empty list will return and it's good.

Reply:
This is a good point -- stop task doesn't check instance status, so I should just use the filter on tasks.